agent/php_error.c

```diff
@@ -29,6 +31,98 @@ typedef struct _nr_php_exception_filter_t {
   nr_php_exception_filter_fn fn;
 } nr_php_exception_filter_t;
 
+/* Transient macro to free memory in nr_php_error_call_error_group_callback */
+#define FREE_MEM                    \
+  nr_php_zval_free(&txn_arr);       \
+  nr_php_zval_free(&error_arr);     \
+  nr_php_zval_free(&group_name_zv); \
+  nro_delete(agent_attributes);     \
+  nr_free(request_uri);             \
+  nr_free(path);                    \
+  nr_free(method);
+
+/*
+ * Purpose      Execute a user-defined PHP callback function that assigns a
+ *              custom group name to an error.
+ *
+ * @param       txn     The transaction object
+ * @param       klass   The error class string
+ * @param       message The error message string
+ * @param       file    The error file string
+ * @param       stack_json  The JSON stack trace string
+ */
+static void nr_php_error_call_error_group_callback(nrtxn_t* txn,
+                                                   char* klass,
+                                                   char* message,
+                                                   char* file,
+                                                   char* stack_json) {
+  zval* txn_arr = NULL;
+  zval* error_arr = NULL;
+  zval* group_name_zv = NULL;
+  zend_fcall_info fci;
+  zend_fcall_info_cache fcc;
+  nrobj_t* agent_attributes;
+  char* group_name_str = NULL;
+  char* request_uri = NULL;
+  char* path = NULL;
+  char* method = NULL;
+  int status_code = 0;
+
+  if (!is_error_callback_set()) {
+    return;
+  }
+
+  agent_attributes = nr_attributes_agent_to_obj(txn->attributes,
+                                                NR_ATTRIBUTE_DESTINATION_ALL);
+
+  request_uri = nr_strdup(nr_php_get_server_global("REQUEST_URI" TSRMLS_CC));
+  path = nr_strdup(txn->path);
+  method = nr_strdup(
+      nro_get_hash_string(agent_attributes, "request.method", NULL));
+  status_code = nr_php_http_response_code();
+  status_code = (status_code < 0) ? 0 : status_code;
+
+  txn_arr = nr_php_zval_alloc();
+  error_arr = nr_php_zval_alloc();
+
+  array_init(txn_arr);
+  array_init(error_arr);
+
+  nr_php_add_assoc_string(txn_arr, "request_uri", request_uri);
+  nr_php_add_assoc_string(txn_arr, "path", path);
+  nr_php_add_assoc_string(txn_arr, "method", method);
```

Comment:
Are these null safe functions?

Reply:
Since they are handled by `nr_strdup`, the value will never be `NULL` for these variables- if the value is null, `nr_strdup` returns an empty string, which is what would be used here.